apps/landing/features/home/HeadingSection/SignUpModal/InputDropdown/Dropdown.tsx

```diff
@@ -0,0 +1,93 @@
+import { type FC } from 'react';
+import { ScrollView, StyleSheet, TouchableOpacity } from 'react-native';
+import { modalActions, Text, View } from '@walless/gui';
+import { ChevronUp } from '@walless/icons';
+
+interface Props {
+	containerWidth: number;
+	optionList: string[];
+	setSelectedOption: (option: string) => void;
+	selectedOption: string;
+	setIsDropped: (isDropped: boolean) => void;
+}
+
+const Dropdown: FC<Props> = ({
+	containerWidth,
+	optionList,
+	selectedOption,
+	setSelectedOption,
+	setIsDropped,
+}) => {
+	const selectedStyle = {
+		backgroundColor: '#1F2A34',
+	};
+
+	return (
+		<View style={{ width: containerWidth }}>
+			<View style={styles.selectedOptionContainer}>
+				<Text>
+					{selectedOption === 'Select one' ? 'Select one...' : selectedOption}
+				</Text>
+				<TouchableOpacity onPress={() => setIsDropped(false)}>
+					<ChevronUp color="#43525F" size={20} />
+				</TouchableOpacity>
+			</View>
+			<ScrollView
+				style={styles.dropdownContainer}
+				showsVerticalScrollIndicator={false}
+			>
+				<Text style={[styles.optionContainer, { color: '#566674' }]}>
+					Select one
+				</Text>
+				{optionList.map((option, index) => (
+					<TouchableOpacity
+						key={index}
+						style={[
+							styles.optionContainer,
+							selectedOption === option && selectedStyle,
+						]}
+						onPress={() => {
+							setSelectedOption(option);
+							setIsDropped(false);
+							modalActions.hide('dropdown');
+						}}
```

Comment:
try separating logic and render by creating 1 function do all these step and pass that only function to `onPress`